tools/loader/ImageClassifier.cpp

```diff
@@ -469,18 +565,43 @@ int main(int argc, char **argv) {
       inputImageData.convertToType(ElemKind::Float16Ty);
     }
 
+    // If we are benchmarking we are done with the while loop.
+    if (iterationsOpt) {
+      break;
+    }
     // About to run inference, so update the input image Placeholder's backing
     // Tensor with inputImageData.
     updateInputPlaceholders(bindings, {inputImagePH}, {&inputImageData});
 
     // Perform the inference execution, updating SMT.
     auto batchSize = inputImageData.dims()[0];
     loader.runInference(bindings, batchSize);
-
     // Print the top-k results from the output Softmax tensor.
     numErrors += processAndPrintResults(SMT, inputImageBatchFilenames);
   }
 
+  if (iterationsOpt) {
+    // Image tensors loaded up to be run at once for benchmark mode.
+    std::vector<std::unique_ptr<Tensor>> preLoadedImages;
+    std::vector<std::unique_ptr<ExecutionContext>> contexts;
+    setupContextPool(outputPH, inputImagePH, inputImageData, preLoadedImages,
```

Comment:
nit: You could probably just return the `std::vector<std::unique_ptr<ExecutionContext>>` from `setupContextPool()` instead of taking it as a reference.